### PR TITLE
Enable `noUncheckedIndexedAccess`

### PR DESCRIPTION
**Pull request by samchungy**
Release seek-tsconfig 2.0

### CODE REVIEW

File: src/cli/test/reporters/github/index.test.ts [hunk 1]
```diff
@@ -120,14 +120,16 @@ const successResults: AggregatedResult = {
...
+  testResults: failResults.testResults[0]
+    ? [
+        {
+          ...failResults.testResults[0],
+          numFailingTests: 0,
+          numPassingTests: 1,
+          testResults: [],
+        },
+      ]
+    : [],
```
Comment (samchungy) [inline src/cli/test/reporters/github/index.test.ts:132]: This isn't quite equivalent (will revisit)

File: .changeset/healthy-mirrors-look.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
+---
+'skuba': minor
```
Comment (samchungy) [inline .changeset/healthy-mirrors-look.md:2]: minor or major? 🤔

File: .changeset/healthy-mirrors-look.md
Comment (72636c) [inline .changeset/healthy-mirrors-look.md:1]: Oof, I'm kinda hot and cold on this change now that the tsconfig-seek upgrade has propagated to a couple of our repos and I'm seeing this PR as well 😅. It seems like there are many common scenarios that are not inferred. For example:

```typescript
const result = /^([a-z])/.match(input);

if (result) {
  // !!!
  result[1];
}

// !!!
''.split('#')[0];

const xs: Array<{key: string, value: string}>;

if (xs.length === 1) {
  // !!!
  const x = xs[0];

  const record = Object.fromEntries(xs.map(x) => [x.key, x.value]);

  // !!!
  record[x.key];
}
```

I don't blame TypeScript here but I worry about the DX on generally well-behaved codebases once we enforce this option. We need to make sure that we can somehow deliver this _without_ normalising usage of easy escape hatches like [`!`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator); what we don't want is for this change to end up proliferating unsafe practices, and beyond just indexed access.

File: template/lambda-sqs-worker/src/app.ts [hunk 3]
```diff
@@ -36,11 +36,17 @@ export const handler = createHandler<SQSEvent>(async (event) => {
 
   const record = event.Records[0];
 
+  const body = record?.body;
+
+  if (!body) {
+    throw new Error(`Record: ${JSON.stringify(record)} is malformed`);
+  }
```
Comment (72636c) [inline template/lambda-sqs-worker/src/app.ts:43]: I'm a bit cautious of logging the whole `record`. Could we do something like:

```suggestion
  if (!record) {{
    throw new Error('Malformed SQS event with no records');
  }
  
  // Or just refer to `record.body` below
  const { body } = record;
```

File: src/cli/init/getConfig.ts [hunk 4]
```diff
@@ -156,6 +156,10 @@ const baseToTemplateData = async ({
 
   const port = String(await getRandomPort());
 
+  if (!orgName) {
+    throw new Error('Could not determine an ownerName');
```
Comment (72636c) [inline src/cli/init/getConfig.ts:160]:
```suggestion
    throw new Error(`Invalid format for owner name: ${ownerName}`);
```

File: src/cli/configure/getEntryPoint.ts [hunk 5]
```diff
@@ -42,6 +42,10 @@ export const getEntryPoint = ({
       // Support exported function targeting, e.g. `src/module.ts#callMeMaybe`
       const [modulePath] = value.split('#', 2);
 
+      if (!modulePath) {
+        return `${chalk.bold(value)} is an invalid module path`;
+      }
```
Comment (72636c) [inline src/cli/configure/getEntryPoint.ts:47]: Is this even possible? The only way I've been able to arrive at an empty array is by doing `''.split('')`.

File: src/api/git/reset.int.test.ts [hunk 6]
```diff
@@ -44,7 +44,7 @@ describe('soft', () => {
 
     const directory = await fs.promises.readdir(dir);
 
-    expect(commits[0].oid).toEqual(initialCommit);
+    expect(commits[0]?.oid).toEqual(initialCommit);
```
Comment (72636c) [inline src/api/git/reset.int.test.ts:47]: It's unfortunate that this applies to tests too; ideally we'd have an ESLint rule so we could disable it on test files.

It's a bit of a wash in this example, but I'd be open to use of `!` in tests. A runtime error is harmless here and would equally indicate that the test failed.

File: scripts/package.ts [hunk 7]
```diff
@@ -171,8 +171,12 @@ const compileChangesByTemplate = (changelog: string) => {
...
         // Note that the changeset entry applies to the template if it existed
         // as of this version and the scope is a match.
-        if (semver.gt(version, added) && templateMatcher.test(templateName)) {
-          changesByTemplate[templateName].push(
+        if (
+          semver.gt(version, added) &&
+          templateMatcher.test(templateName) &&
+          changesByTemplate[templateName]
+        ) {
+          changesByTemplate[templateName]!.push(
```
Comment (72636c) [inline scripts/package.ts:179]:
```suggestion
          changesByTemplate[templateName].push(
```

**Review by 72636c — APPROVED** [review]

**Review by 72636c — APPROVED** [review]

**Pull request merged**